src/core/public/http/http_service.ts

```diff
@@ -17,84 +17,31 @@
  * under the License.
  */
 
-import * as Rx from 'rxjs';
-import {
-  distinctUntilChanged,
-  endWith,
-  map,
-  pairwise,
-  startWith,
-  takeUntil,
-  tap,
-} from 'rxjs/operators';
-
-import { Deps } from './types';
-import { setup } from './fetch';
+import { HttpDeps, IHttpService } from './types';
+import { setup } from './http_setup';
 
 /** @internal */
 export class HttpService {
-  private readonly loadingCount$ = new Rx.BehaviorSubject(0);
-  private readonly stop$ = new Rx.Subject();
-
-  public setup(deps: Deps) {
-    const { fetch, shorthand } = setup(deps);
-
-    return {
-      fetch,
-      delete: shorthand('DELETE'),
-      get: shorthand('GET'),
-      head: shorthand('HEAD'),
-      options: shorthand('OPTIONS'),
-      patch: shorthand('PATCH'),
-      post: shorthand('POST'),
-      put: shorthand('PUT'),
-      addLoadingCount: (count$: Rx.Observable<number>) => {
-        count$
-          .pipe(
-            distinctUntilChanged(),
-
-            tap(count => {
-              if (count < 0) {
-                throw new Error(
-                  'Observables passed to loadingCount.add() must only emit positive numbers'
-                );
-              }
-            }),
+  private satupService!: IHttpService;
+  private startedService!: IHttpService;
```

Comment:
With two instances, doesn't this mean we have two different loading counts? I think for stateful services like this, it's fine to expose the same instance so we don't lose state from *setup* once *start* begins.

Reply:
That is correct. I wasn't sure if we would need different counts per phase, but I can move this to the outer scope to resolve that.